File: c++/Solution252.cpp
//Intervals O(nlogn) O(n), sort需要O(nlogn) O(n)
class Solution252 {//lintcode920
public:
    bool canAttendMeetings(vector<Interval> &intervals) {
        sort(intervals.begin(), intervals.end(), [](Interval& a, Interval& b){
                return a.start < b.start;//依照左界小排到大
            });

        for(int i=1; i<intervals.size(); ++i){
            if(intervals[i - 1].end > intervals[i].start)
                return false;
        }
        return true;
    }
};

File: java/Solution252.java
package leetCode.java;

import java.util.*;

//Intervals O(nlogn) O(n), sort需要O(nlogn) O(n)
class Solution252 {
    public boolean canAttendMeetings(List<Interval> intervals) {
        Collections.sort(intervals, (a, b) -> a.start - b.start);

        for (int i = 1; i < intervals.size(); ++i) {
            // 重疊條件: a.end < b.start && a.start < b.end
            // a.start < b.end在排序時已經確定
            // 不排序的話就要所有intervals互相比較, 時間O(n^2)更慢
            if (intervals.get(i).start < intervals.get(i - 1).end) {
                return false;
            }
        }
        return true;
    }
}